trip-schema/src/main/java/global/maplink/trip/schema/v2/problem/TollRequest.java

```diff
@@ -3,24 +3,157 @@
+import static java.util.Objects.isNull;
 import static lombok.AccessLevel.PRIVATE;
 
 @Data
 @Builder
 @RequiredArgsConstructor
 @NoArgsConstructor(force = true, access = PRIVATE)
-public class TollRequest {
+public class TollRequest implements Validable {
+    private static final String SITE_NOT_USED = "site not used";
+    private static final String SITE_USED_AS_FROM_SITE_ID = "site used as fromSiteId";
+    private static final String SITE_USED_AS_TO_SITE_ID = "site used as toSiteId";
+    private static final String SITE_USED_AS_MIDDLE_SITE = "site used as middle site";
+
     private final TollVehicleType vehicleType;
-    private final Billing billing;
+    @Builder.Default
+    private final Billing billing = Billing.DEFAULT;
     @Builder.Default
     private final Set<TransponderOperator> transponderOperators = new HashSet<>(Collections.singletonList(TransponderOperator.SEM_PARAR));
+    private final List<LegVariableAxles> variableAxles;
+
+    @Override
+    public List<ValidationViolation> validate() {
+        List<ValidationViolation> errors = new ArrayList<>();
+        if (vehicleType == null) {
+            errors.add(TOLL_PARAMETERS_DOES_NOT_HAVE_VEHICLE_TYPE);
+        }
+        return errors;
+    }
+
+    public List<ValidationViolation> validateVariableAxles(final List<SitePoint> sites){
+        List<ValidationViolation> errors = new ArrayList<>();
+
+        if (variableAxles == null || sites.isEmpty()) {
+            return errors;
+        }
+
+        List<String> problemSites = getProblemSites(sites);
+        Map<String, String> sitesStatusMap = createVirginSitesStatusMap(problemSites);
+
+        for (LegVariableAxles legVariableAxles : variableAxles) {
+            String fromSiteId = legVariableAxles.getFromSiteId();
+            String toSiteId = legVariableAxles.getToSiteId();
+
+            if (isNull(fromSiteId)){
+                errors.add(VARIABLE_AXLES_FROMSITEID_EMPTY);
+                return errors;
+            }
+
+            if (!problemSites.contains(fromSiteId)) {
+                errors.add(MSG_SITEID_NOT_FOUND_IN_PROBLEM.addToMessage(fromSiteId));
+            }
+
+            if (isNull(toSiteId)){
+                errors.add(VARIABLE_AXLES_TOSITEID_EMPTY);
+                return errors;
+            }
+
+            if (!problemSites.contains(toSiteId)) {
+                errors.add(MSG_SITEID_NOT_FOUND_IN_PROBLEM.addToMessage(toSiteId));
+            }
+
+            if (fromSiteId.equalsIgnoreCase(problemSites.get(problemSites.size() - 1))) {
+                errors.add(VARIABLE_AXLES_FROMSITEID_POINTING_TO_LAST_SITE);
+                return errors;
+            }
+
+            if (fromSiteId.equalsIgnoreCase(toSiteId)) {
+                errors.add(VARIABLE_AXLES_FROMSITEID_SAME_AS_TOSITEID.addToMessage(fromSiteId));
+                return errors;
+            }
+
+            List<String> TOLL_VEHICLE_TYPES = Arrays.stream(TollVehicleType.values())
```

Comment:
variável local em geral usamos camelCase, essa sintaxe é mais para uma constante.
Poderia mapear esse campo para fora do for, para não ficar reexecutando esse mesmo código pra cada iteração do for

Reply:
ok, vou mudar aqui...valeu.